prediction_market_agent/agents/arbitrage_agent/deploy.py

```diff
@@ -0,0 +1,211 @@
+        filter_by: FilterBy = FilterBy.OPEN,
+    ) -> t.Sequence[AgentMarket]:
+        return super().get_markets(
+            market_type=market_type,
+            limit=50,
+            sort_by=SortBy.HIGHEST_LIQUIDITY,
+            # Fetching most liquid markets since more likely they will have related markets
+            filter_by=FilterBy.OPEN,
+        )
+
+    def answer_binary_market(self, market: AgentMarket) -> ProbabilisticAnswer | None:
+        return ProbabilisticAnswer(p_yes=Probability(0.5), confidence=1.0)
+
+    def _build_chain(self) -> RunnableSerializable[t.Any, t.Any]:
+        llm = ChatOpenAI(
+            temperature=0,
+            model=self.model,
+            api_key=APIKeys().openai_api_key_secretstr_v1,
+        )
+
+        parser = PydanticOutputParser(pydantic_object=Correlation)
+        prompt = PromptTemplate(
+            template=PROMPT_TEMPLATE,
+            input_variables=["main_market_question", "related_market_question"],
+            partial_variables={"format_instructions": parser.get_format_instructions()},
+        )
+
+        chain = prompt | llm | parser
+        return chain
+
+    @observe()
+    def calculate_correlation_between_markets(
+        self, market: AgentMarket, related_market: AgentMarket
+    ) -> Correlation:
+        correlation: Correlation = self.chain.invoke(
+            {
+                "main_market_question": market.question,
+                "related_market_question": related_market.question,
+            }
+        )
+        return correlation
+
+    @observe()
+    def get_correlated_markets(self, market: AgentMarket) -> list[CorrelatedMarketPair]:
+        # We try to find similar, open markets which point to the same outcome.
+        correlated_markets = []
+        # We only wanted to find related markets that are open.
+        # We intentionally query more markets in the hope it yields open markets.
+        # We could store market_status (open, closed) in Pinecone, but we refrain from it
+        # to keep the chain data (or graph) as the source-of-truth, instead of managing the
+        # update process of the vectorDB.
+
+        related = self.pinecone_handler.find_nearest_questions_with_threshold(
+            limit=self.max_related_markets_per_market,
+            text=market.question,
+            filter={
+                "close_time_timestamp": {"gte": utcnow().timestamp() + 3600}
+            },  # closing 1h from now
+        )
+
+        omen_markets = self.subgraph_handler.get_omen_binary_markets(
+            limit=len(related),
+            id_in=[i.market_address for i in related],
+            resolved=False,
+        )
+
+        # Order omen_markets in the same order as related
+        related_market_addresses = [i.market_address for i in related]
+        omen_markets = sorted(
+            omen_markets, key=lambda m: related_market_addresses.index(m.id)
+        )
+        omen_markets = [m for m in omen_markets if m.id != market.id]
```

Comment:
This can be done directly in `id_in=[i.market_address for i in related]` above.

Reply:
Thanks, updated the lambda expression.